### PR TITLE
Add support for assembly submissions

### LINKED ISSUE #2: Add support for assembly submissions
I'm sure there are many people who would be interested in submitting in assembly, especially during a contest. This should be honored and should be very easy to implement.

### PR DESCRIPTION
**Pull request by wdotmathree**
This WILL break especially with all the refactoring in sedimentation. Assembly support should be disabled in production by removing the dropdown option. Very minimal stress testing was done. Use at your own risk.

### CODE REVIEW

**Comment by kevlu8** [comment]
you literally added your assembler and no other changes

**Comment by wdotmathree** [comment]
which is why its work in progress

**Comment by kevlu8** [comment]
u right u right mb

**Comment by wdotmathree** [comment]
i forgot to push because i was having major issues with git yesterday

**Comment by wdotmathree** [comment]
and im on windows rn so ill push later

**Comment by wdotmathree** [comment]
my nvidia drivers broke so cannot push

**Comment by wdotmathree** [comment]
drivers are fixed

File: judging-backend/main.cpp [hunk 1]
```diff
@@ -215,6 +215,69 @@ int main(int argc, char *argv[]) {
...
+			// child process
+			struct rlimit rlim;
+			rlim.rlim_cur = 5;
+			rlim.rlim_max = 5;
+			if (setrlimit(RLIMIT_CPU, &rlim)) {
+				std::cerr << "failed to set compiler time limit" << std::endl;
+				return IE;
+			}
+
+			std::string asm_filepath = dir + "/../../sedimentation-assembler/sedimentation";
```
Comment (kevlu8) [inline judging-backend/main.cpp:232]: what is this...

**Review by kevlu8 — COMMENTED** [review]
Overall looks decent, just make sure you do some more testing to make sure that Assembly is fully functional. From what I see so far, there is no reason that it should fail during the testing phase, but hardcoding your assembler into the source code is suboptimal.

File: judging-backend/main.cpp (hunk 1 above)
Comment (wdotmathree) [reply judging-backend/main.cpp:232]: It's a reliable way to reference sedimentation in the submodule without needing the user to install the executable into their system.

**Comment by wdotmathree** [comment]
> Overall looks decent, just make sure you do some more testing to make sure that Assembly is fully functional. From what I see so far, there is no reason that it should fail during the testing phase, but hardcoding your assembler into the source code is suboptimal.

The user is free to choose whether to download the submodule containing my assembler when cloning the repository. They are also free to change the assembler path in the code. A potential future improvement is to move all the compiler paths/arguments to the top of the file to facilitate easy substitution.

**Comment by wdotmathree** [comment]
A hotfix will be pushed shortly to sedimentation to make it accept `-f elf64` as an alias to `-f elf` in order to make it easier to change it to nasm without needing to edit the arguments. This should NOT be merged until a commit updating the submodule with this hotfix is added.

I assign this change 0.1 story points and a T-shirt size of XXS, however this must be delayed until tomorrow as I have to attend 8hrs of meetings with the project manager followed by another 8hrs of meetings with the client. This PR should be ready to merge by 20:00 UTC-06:00 tomorrow.

File: judging-backend/main.cpp (hunk 1 above)
Comment (kevlu8) [reply judging-backend/main.cpp:232]: How can we guarantee that sedimentation-assembler/sedimentation exists? We could add a Makefile to compile the judge and sedimentation, if you're up to doing that.

**Comment by kevlu8** [comment]
In my opinion, this change actually requires closer to $e^{\sqrt{0.231}}-\sqrt{2}$ story points. Would it be possible to have this addressed ASAP?

**Comment by wdotmathree** [comment]
In the interest of developer sanity, no.

**Comment by wdotmathree** [comment]
No.

**Comment by kevlu8** [comment]
Fuck you

**Comment by kevlu8** [comment]
I'm a kind soul so this time I won't merge it and brick the master branch of our repo.

**Comment by wdotmathree** [comment]
I will

**Comment by kevlu8** [comment]
WHATTTTTT

**Comment by niinpatel** [comment]
🍿 🍿

**Comment by EXtremeExploit** [comment]
we need part 3 of this crazy conversation, cant leave us hanging like that...

**Comment by kevlu8** [comment]
stay tuned!!!

**Comment by wdotmathree** [comment]
um i think that was actually a bad idea since i cant push my hotfix anymore

**Comment by kevlu8** [comment]
you ACTUALLY bricked the master branch????? ohhh my days

**Comment by Binozo** [comment]
What even is this 💀

**Comment by Binozo** [comment]
> my nvidia drivers broke so cannot push

image

**Comment by enganese** [comment]
Guys, why the fuck are we using github pr's as a messenger 💀

**Comment by sandevins** [comment]
It's how is meant to be used

**Comment by dwisiswant0** [comment]
> my nvidia drivers broke so cannot push

image

**Comment by araluce** [comment]
when is the next episode? 🍿

**Comment by includesec-erik** [comment]
....................../´¯/) 
....................,/¯../ 
.................../..../ 
............./´¯/'...'/´¯¯`·¸ 
........../'/.../..../......./¨¯\ 
........('(...´...´.... ¯~/'...') 
.........\.................'...../ 
..........''...\.......... _.·´ 
............\..............( 
..............\.............\...

Fuk da police

**Comment by hiramamador** [comment]
Are we still commenting after closure?

**Comment by quick007** [comment]
somehow this became a meme

image

**Comment by elijah629** [comment]
I'm hyped for season 4

**Comment by RA341** [comment]
Someone get Netflix I need a shitty live action adaptation for this.

**Comment by 3dcantaloupe** [comment]
This is awesome hahhaa

**Comment by electricsteve** [comment]
yea holy shit this is fun to see

**Comment by chris-amaya** [comment]
whoever reads this 

fuck you 🖕

**Comment by User8395** [comment]
To be continued

**Comment by polyzium** [comment]
I just discovered this repository, WTH is this?

**Comment by AliAryanTech** [comment]
Why you all got so much time to scroll memes?

**Comment by IlyaNavodkin** [comment]
Пацаны здарова у меня винда не работает - это из за вас?

**Comment by User8395** [comment]
> Пацаны здарова у меня винда не работает - это из за вас?

"Hi guys, my Windows isn't working - is it because of you?"

**Comment by Willie169** [comment]
When is the next season?

**Comment by tamilvanan-bn** [comment]
Here from a Discord meme 😂

**Comment by hexett** [comment]
fireship discord head ass

**Comment by Sanix-Darker** [comment]
are this guys still up and running ?
We need a seQuel.

**Comment by satyaorz** [comment]
📎

**Pull request merged**